spec/src/stdlib.makam

```diff
@@ -0,0 +1,22 @@
+% Extensions to Makam's standard library
+
+%extend map.
+
+    keys : map K V -> set K -> prop.
+    keys [] [].
+    keys ((Key, _) :: Map) (Key :: Keys) :-
+        keys Map Keys.
+
+%end.
+
+
+%extend set.
+
+    map : (A -> prop) -> set A -> prop.
+    map P [].
+    map P (Elem :: Set) :-
+        P Elem,
+        set.remove_if_member Set Elem Set',
+        map P Set.
+
+%end.
```

Comment:
Wondering if these extensions would be handy to have in the standard library @astampoulis?

Reply:
Sure, these would be useful if you feel like opening a PR!

BTW I think the second line is not being used as is, do you mean `map P Set'` in the last line maybe?